docs/using-the-jdbc-driver/using-plugins/UsingTheFailover2Plugin.md

```diff
@@ -0,0 +1,76 @@
+The AWS Advanced JDBC Driver uses the Failover Plugin v2 to provide minimal downtime in the event of a DB instance failure. The plugin is a next version (v2) of [Failover Plugin](./UsingTheFailoverPlugin.md) and most of the explanations and suggestions for Failover Plugin are applicable to Failover Plugin v2 (unless opposite is explicitly stated).
+
+## Differences between Failover Plugin and Failover Plugin v2
+
+Failover Plugin performs a failover process for each DB connection, and it's triggered independently and unrelated to failover processes in other connections. While such independence between failover processes bring some benefits, it also leads to additional resources like extra threads. If dozens of DB connections are being failed over at the same time, it may cause significant load on a client environment.
+
+<div style="text-align:center"><img src="../../images/failover1.jpg" /></div>
+<i>Picture 1. Each connection triggers its own failover process to detect a new writer.</i>
+<br><br>
+
+If connection needs to get a latest topology, it calls `RdsHostListProvider`. It should be noted that `RdsHostListProvider` runs in the same thread as a connection failover process. As shown on _Picture 1_ above, different connections start and end their failover processes independently.
+
+Failover Plugin v2 uses an optimized approach when a process of detecting and confirming of a cluster topology is delegated to central topology monitoring component that runs in a separate thread. When topology is confirmed and a new writer is detected, each of waiting connection can resume and reconnect to a required node. Such design helps to minimize resources required for failover processing. It scales better comparing to Failover Plugin.
+
+<div style="text-align:center"><img src="../../images/failover2.jpg" /></div>
+<i>Picture 2. Connections call MonitoringRdsHostListProvider and delegate it a new writer detection. While waiting for MonitoringRdsHostListProvider, connection threads suspend.</i>
+<br><br>
+
+If two connections get communication issue with their internal (physical) DB connections, each connection may request a topology monitoring component (`MonitoringRdsHostListProvider` on a _Picture 2_) to provide a new writer. Both connections are notified as soon as a latest topology gets available. Connection threads can resume and continue with their suspended workflows and to reconnect to a reader or a writer node. 
+
+The topology monitoring component mentioned above (`MonitoringRdsHostListProvider`) updates topology periodically. Usually it uses a connection to a writer node to fetch a cluster topology. Using a connection to a writer node allows to get topology from the first hand without a risk of getting stale data as in case of fetching it from a reader. In some exceptional cases the monitoring component may (temporarily) use a reader connection to fetch a topology however it switch back to a writer node as soon as possible.
```

Comment:
```suggestion
The topology monitoring component mentioned above (`MonitoringRdsHostListProvider`) updates the topology periodically. Usually it uses a connection to a writer node to fetch the cluster topology. Using a connection to a writer node helps the component retrieve the topology without the risk of getting stale data, which may occur if fetching the topology from a reader. In some exceptional cases the monitoring component may temporarily use a reader connection to fetch the topology, but it will switch back to a writer node as soon as possible.
```